src/twisted/application/twist/test/test_options.py

```diff
@@ -117,6 +117,20 @@ def test_installReactor(self):
         self.assertEqual(set(self.installedReactors), set(["fusion"]))
 
 
+    def test_installCorrectReactor(self):
+        """
+        L{TwistOptions.installReactor} installs the chosen reactor after the
+        command line options have been parsed.
+        """
+        self.patchInstallReactor()
+
+        options = TwistOptions()
+        options.subCommand = "test-subcommand"
+        options.parseOptions(["--reactor=fusion"])
+
+        self.assertEqual(set(self.installedReactors), set(["fusion"]))
```

Comment:
Great!

One thing about the way all of this code is being tested is that, instead of monkey-patching the options object, it could probably be using `twisted.internet.test.modulehelpers.NoReactor`.  This is a helper that basically monkey-patches the `twisted.internet.reactor` module.  I think this would isolate the tests a bit more from the implementation and go a step further in asserting the thing we actually want to assert (that the selected reactor is installed).  A possible downside is that I don't think you'll be able to install the "fusion" reactor this way (since no such thing exists at the moment :smile: ).

This is not a blocking comment for this PR.  It's more like a suggestion for a follow-up improvement of this test suite.